src/Uno.UI/UI/Xaml/Controls/ContentControl/ContentControl.cs

```diff
@@ -119,6 +119,10 @@ public virtual object Content
 				typeof(ContentControl),
 				new FrameworkPropertyMetadata(
 					defaultValue: null,
+					// Don't propagate DataContext to Content qua Content, only propagate it via the visual tree. Prevents spurious
+					// propagation in case that default style and template is only applied once the control enters the visual tree
+					// (ie if created in code by new SomeControl())
+					FrameworkPropertyMetadataOptions.ValueDoesNotInheritDataContext,
```

Comment:
Looks like we should be having DataContext or Bindings changes count unit tests, let's see if that fails first :)

Reply:
All the existing tests seem to pass at least!

Reply:
Added a test which counts DC and binding updates